tipg/middleware.py

```diff
@@ -51,19 +51,38 @@ async def dispatch(self, request: Request, call_next):
 class CatalogUpdateMiddleware(BaseHTTPMiddleware):
     """Middleware to update the catalog cache."""
 
+    def __init__(
+        self,
+        app: ASGIApp,
+        ttl: int = 300,
+    ) -> None:
+        """Init Middleware.
+
+        Args:
+            app (ASGIApp): starlette/FastAPI application.
+            ttl (int): time-to-live value in seconds. Defaults to 300.
+
+        """
+        super().__init__(app)
+        self.ttl = ttl
+
     async def dispatch(self, request: Request, call_next):
         """Fetch Catalog either immediately or in background."""
-        if request.query_params.get("refresh"):
-            logger.debug("Forcing catalog refresh.")
-            await register_collection_catalog(request.app)
-            response = await call_next(request)
-        elif datetime.now() > request.app.state.catalog_expire:
+        collection_catalog = getattr(request.app.state, "collection_catalog", {})
+        collections = collection_catalog.get("collections")
+        last_updated = collection_catalog.get("last_updated")
+
+        if not collections or datetime.now() > (
+            last_updated + timedelta(seconds=self.ttl)
+        ):
```

Comment:
2 cases:
- there is not collections: we update (🤷‍♂️) 
- collections are too old: we update

Reply:
changed this to only rely on the `last_updated` tag. 

Starting the application with a brand new database could in theory mean that every request would try to refresh the catalog, which IMO seems a bit risky

Reply:
yeah but what if the `startup` fails or if there is not collections at all 🤷 

I think it's fine as is right now!